internal/utils/utils.go

```diff
@@ -27,14 +28,12 @@ var seededRand = rand.New(rand.NewSource(time.Now().UnixNano()))
 
 // GetArchitecture retrieves if the underlying system platform is arm64 or amd64
 func GetArchitecture() string {
-	arch := "amd64"
-
-	envArch := os.Getenv("GOARCH")
-	if envArch == "arm64" {
-		arch = "arm64"
+	arch, present := os.LookupEnv("GOARCH")
```

Comment:
Should we read from the env first or directly get the value from runtime?

Reply:
So for pure vm's we can pull from the runtime, for things like autodiscover which uses docker we may need to still pull that env var?

Reply:
I'm not sure if we even use this method elsewhere than fleet 🤔